### DISCUSSION TITLE
Should we use Conventional Commits?

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#2575] [General]
We have had various discussions/attempts at introducing tooling to help create changelogs, as a step towards that should we start using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)?

**Comment by apoelstra** [comment]
Yeah, I'd be willing to give it a shot. Since none of us are familiar with it I guess we'll have to try policing each other.

**Reply by tcharding** [reply]
`chore`

**Comment by sanket1729** [comment]
Not familiar with it. But open to experiment as a soft(non-enforcing) policy.

**Comment by dpc** [comment]
Warning a while, you'll be writting `chore: ` in front of everything.

**Reply by tcharding** [reply]
I hit that already, and I'm not super keen on calling arbitrary improvements 'chores', I love programming but I'm not that fond of doing chores - seems incongruous.

**Comment by tcharding** [comment]
I guess you guys are using it in Fedimint @dpc - in hindsight was it a good idea?

**Reply by dpc** [reply]
We don't have the will to be more careful about `chore/feat/refactor` distinction, even less so to scope field, so I feel like most stuff is just a `chore`, and all in all it doesn't really give us anything, other than just forcing lower case letters.

We might need to up our game in the future, but RN there's so much fundamental and important things to get done quickly, that stuff like that is not very important.

On the positive side - it's not a lot of effort, and doesn't feel cumbersome.

**Comment by tcharding** [comment]
I've already found the `fix: ` and `feat: ` prefixes useful in enabling looking back at a list of patches to see what semver change needs making when doing release PRs.

Until we work out exactly what tool we are going to use to autogenerate changelog notes I can't see the benefit in putting `chore: ` before everything.

**Reply by tcharding** [reply]
Note that that is also what our github labels are meant to do but I find it easier to look and the git index in the terminal that using github.

**Comment by Kixunil** [comment]
I'm not fond of it since it takes up valuable space in commit summary. If we need metadata it should go at the end of the commit message.

**Comment by tcharding** [comment]
Oh I forgot this was open, I (we?) tried for about 2 days and found it totally useless.
